azure-iot-device/azure/iot/device/provisioning/provisioning_device_client.py

```diff
@@ -0,0 +1,40 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module provides an abstract interface representing clients which can communicate with the
+Device Provisioning Service.
+"""
+
+import abc
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class ProvisioningDeviceClient(object):
+    """
+    Super class for any client that can be used to register devices to Device Provisioning Service.
+    """
+
+    def __init__(self, transport):
+        """
+        Initializes the registration client.
+        :param transport: Instance of the Transport object.
+        """
+        self._transport = transport
+
+    @abc.abstractmethod
+    def register(self):
+        """
+        Register the device with the Device Provisioning Service.
+        """
+        pass
+
+    @abc.abstractmethod
+    def cancel(self):
+        """
+        Cancel an in progress registration of the device with the Device Provisioning Service.
```

Comment:
>Service [](start = 86, length = 7)

This comment should also specify exactly what happens when you "cancel" a registration.  Specifically, does it do something to send a "cancel command" to the provisioning service, or does it just stop waiting for a response?  Is there any confirmation from the provisioning server that any kind of cancellation actually happened?  Does it disconnect the transport after cancelling?  Does it wait for the disconnect to complete before returning?  What happens to the original call to register?  What happens if the user calls cancel, but they never called register in the first place, or they call cancel after the registration is complete, or they call cancel after the registration has already been cancelled?  What state does this leave the client in? (i.e.  Can the user try to re-register after cancelling)?  We both know the answers to these questions, but, a good rule-of-thumb is "if you asked the question while implementing this, it probably belongs in the doc strings that the caller will see".